### PR TITLE
USE Mavlink typemask

### PR DESCRIPTION
**Pull request by khancyr**
Rework from https://github.com/ArduPilot/ardupilot/pull/15986
close https://github.com/ArduPilot/ardupilot/pull/15986

This is cover by autotest

### CODE REVIEW

File: Rover/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -79,11 +79,15 @@ void GCS_MAVLINK_Rover::send_position_target_global_int()
...
     }
+    static constexpr uint16_t POSITION_TARGET_TYPEMASK_BIT_12 = (1U<<12);
+    static constexpr uint16_t TYPE_MASK = POSITION_TARGET_TYPEMASK_VX_IGNORE | POSITION_TARGET_TYPEMASK_VY_IGNORE | POSITION_TARGET_TYPEMASK_VZ_IGNORE |
+                                          POSITION_TARGET_TYPEMASK_AX_IGNORE | POSITION_TARGET_TYPEMASK_AY_IGNORE | POSITION_TARGET_TYPEMASK_AZ_IGNORE |
+                                          POSITION_TARGET_TYPEMASK_FORCE_SET | POSITION_TARGET_TYPEMASK_YAW_IGNORE| POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE | POSITION_TARGET_TYPEMASK_BIT_12;
     mavlink_msg_position_target_global_int_send(
         chan,
         AP_HAL::millis(), // time_boot_ms
         MAV_FRAME_GLOBAL, // targets are always global altitude
-        0xFFF8, // ignore everything except the x/y/z components
```
Comment (peterbarker) [inline Rover/GCS_Mavlink.cpp:86]:
```
>>> hex(mavutil.mavlink.POSITION_TARGET_TYPEMASK_VX_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_VY_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_VZ_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AX_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AY_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AZ_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_FORCE_SET | mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_IGNORE| mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE | POSITION_TARGET_TYPEMASK_BIT_12)
'0x1ff8'
```

**Review by peterbarker — CHANGES_REQUESTED** [review]
Values must match - until we decide we need to fix these.

I've only checked the one value.

**Review by khancyr — COMMENTED** [review]
```
from pymavlink import mavutil
POSITION_TARGET_TYPEMASK_LAST = 0xF000

hex(mavutil.mavlink.POSITION_TARGET_TYPEMASK_VX_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_VY_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_VZ_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AX_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AY_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AZ_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_FORCE_SET | mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_IGNORE| mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE | POSITION_TARGET_TYPEMASK_LAST) 

Out[14]: '0xfff8'
```


```
In [15]: hex(mavutil.mavlink.POSITION_TARGET_TYPEMASK_VX_IGNORE | mavutil.mav
    ...: link.POSITION_TARGET_TYPEMASK_VY_IGNORE | mavutil.mavlink.POSITION_T
    ...: ARGET_TYPEMASK_VZ_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_
    ...: AX_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AY_IGNORE | mav
    ...: util.mavlink.POSITION_TARGET_TYPEMASK_AZ_IGNORE | mavutil.mavlink.PO
    ...: SITION_TARGET_TYPEMASK_FORCE_SET | mavutil.mavlink.POSITION_TARGET_T
    ...: YPEMASK_YAW_IGNORE| mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_RAT
    ...: E_IGNORE)                                                           
Out[15]: '0xff8'
```

```
In [16]: hex(mavutil.mavlink.POSITION_TARGET_TYPEMASK_X_IGNORE | mavutil.mavl
    ...: ink.POSITION_TARGET_TYPEMASK_Y_IGNORE | mavutil.mavlink.POSITION_TAR
    ...: GET_TYPEMASK_Z_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AX_
    ...: IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_AY_IGNORE | mavuti
    ...: l.mavlink.POSITION_TARGET_TYPEMASK_AZ_IGNORE | mavutil.mavlink.POSIT
    ...: ION_TARGET_TYPEMASK_FORCE_SET | mavutil.mavlink.POSITION_TARGET_TYPE
    ...: MASK_YAW_IGNORE| mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_RATE_I
    ...: GNORE)                                                              
Out[16]: '0xfc7'
```

I```
n [17]: hex(mavutil.mavlink.POSITION_TARGET_TYPEMASK_AX_IGNORE | mavutil.mav
    ...: link.POSITION_TARGET_TYPEMASK_AY_IGNORE | mavutil.mavlink.POSITION_T
    ...: ARGET_TYPEMASK_AZ_IGNORE | mavutil.mavlink.POSITION_TARGET_TYPEMASK_
    ...: FORCE_SET | mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_IGNORE| mav
    ...: util.mavlink.POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE)              
Out[17]: '0xfc0'
```

**Comment by tridge** [comment]
waiting for review confirmation from @peterbarker

**Review by peterbarker — CHANGES_REQUESTED** [review]
Bad rebase?

Needs to compile

**Comment by khancyr** [comment]
corrected !

**Review by peterbarker — APPROVED** [review]
I've done the maths on each of these now and it works out.

**Review by tridge — APPROVED** [review]

**Pull request merged**